src/pip/_internal/commands/install.py

```diff
@@ -446,12 +439,13 @@ def run(self, options, args):
         return SUCCESS
 
     def _handle_target_dir(self, target_dir, target_temp_dir, upgrade):
+        # type: (str, Optional[TempDirectory], bool) -> None
```

Comment:
Why `target_temp_dir` is `Optional` here? AFAICU, it'd be provided for sure, won't it?

Reply:
I was doing this because both `target_dir` and `target_temp_dir` were assigned None by default

https://github.com/pypa/pip/blob/d01bfcfaa13a4f06fa0ce61fa18cf06012f2e78f/src/pip/_internal/commands/install.py#L257-L258

But thinking more on it, we can update the if condition below to also check for `target_temp_dir`

https://github.com/pypa/pip/blob/d01bfcfaa13a4f06fa0ce61fa18cf06012f2e78f/src/pip/_internal/commands/install.py#L441-L444

Which I have added in https://github.com/pypa/pip/pull/8365/commits/8c27d2d3666fc84d1c9fbedd2768b1777d66add1